Component server application

cpaech opened the pull request:
I have now added the first working server / client prototype. This works by having a flag (boolean) in Main.java which toggles between starting the server or the client at the beginning. If the Server is started there is now a ServerController, which gets updated every frame. Incoming new connections are handled by a seperated thread in IncommingConnections.java. Messages can be received by each connected client. Each Message is compromised of 2 distinct features. The ID and the content. All the necessary Structures are highlighted in Messages.md. Meanwhile on the client a connection to the Server is attempted. In the future this should be done via a seperate menu, where the IP can be inputed. Messages can be sent by creating a new class with the correct constructor and then calling the Send method with the neccessary outputstream of the socket.

cpaech committed:
First Working MessageTest

cpaech committed:
MessageDoc, Comments, Error Checking

Added more comments to generate for javadoc.
Added a Message.md which contains documentation for all the types of network messages and their structure.
Added more try-catches and error checks for stability and easier debugging.

cpaech committed:
Removed unused imports

cpaech committed:
removed unused imports

ffeibl reviewed (approved):
The stuff I did have was more or less the same. Code looks very good, maybe a bit overengeneered but I'm not complaining.

cpaech commented:
This is not threadsafe.

cpaech committed:
Threadsafety and error handling

Made the Incoming Sockets exchange thread safe.
Made it so there is no longer a Socket reset error when closing the client -> disposing of ressources

cpaech committed:
Added missing comments

cpaech commented:
New and fixed version
